pandas/core/reshape/pivot.py

```diff
@@ -368,18 +368,34 @@ def _convert_by(by):
 @Appender(_shared_docs['pivot'], indents=1)
 def pivot(data, index=None, columns=None, values=None):
     if values is None:
-        cols = [columns] if index is None else [index, columns]
+        if index is None:
+            cols = [columns]
+        else:
+            if is_list_like(index):
+                # If a given index is a list, set cols to index.
```

Comment:
similar to below can you make more informative comments

Reply:
Any suggestion ? I am still new to these. Thanks.